chromium_src/ui/views/controls/button/md_text_button.h

```diff
@@ -8,26 +8,28 @@
 
 #include "third_party/abseil-cpp/absl/types/optional.h"
 #include "ui/gfx/vector_icon_types.h"
+#include "ui/views/controls/button/label_button.h"
 
 // Rename MdTextButton to MdTextButtonBase
 #define MdTextButton MdTextButtonBase
 
-// Define a Brave-specific method we can get called from UpdateColors() to
-// extend its functionality instead of defining UpdateColors() "virtual" and
-// overriding it in our version of the MdTextButton class because there are some
-// subclasses that define their own UpdateColors() method (OmniboxChipButton)
-// now, which would not work with the virtual + override approach.
-#define UpdateColors                       \
-  UpdateColors_Unused();                   \
+// Define a Brave-specific method to be called from UpdateColors() to extend its
+// functionality instead of defining UpdateColors() "virtual" and overriding it
+// in our version of the MdTextButton class because there are some subclasses
+// that define their own UpdateColors() method (OmniboxChipButton) now, which
+// would not work with the virtual + override approach.
+// Note: We redefine UpdateBackgroundColor because we want it to be protected.
+#define UpdateBackgroundColor              \
```

Comment:
and how about removing `md_text_button.cc.patch` while you're here?
If making `UpdateColors()` as virtual in protected, it could be overridden by ours also like you did for `UpdateBackgroundColor()`? :)

Reply:
I'm going to make a followup PR, where do this + the background split.

Reply:
Actually, we can't remove this, because the `UpdateIconForBrave` method still needs the patch :/

Reply:
@fallaciousreasoning I'm not following. The only patch is for `BRAVE_MD_TEXT_BUTTON_UPDATE_COLORS` which can be removed by making `UpdateColors` virtual as @simonhong is suggesting. You can call the superclass method and then do whatever `BRAVE_MD_TEXT_BUTTON_UPDATE_COLORS` does

Reply:
I got scared off by the comment saying to not make `UpdateColors` virtual. However, I just tested and the `OmniboxChipButtons` look fine, so I guess that issue has been resolved in the 2+ years since the comment. Adding it to my follow up PR https://github.com/brave/brave-core/pull/16124